src/Browser.php

```diff
@@ -725,6 +728,74 @@ public function withResponseBuffer($maximumSize)
         ));
     }
 
+    /**
+     * The `withHeader(string $header, string $value): Browser` method can be
+     * used to add a request header for all following requests.
+     *
+     * ```php
+     * $browser = $browser->withHeader('User-Agent', 'ACME');
+     *
+     * $browser->get($url)->then(function (Psr\Http\Message\ResponseInterface $response) {
+     * //will contain the header User-Agent with the value ACME
+     * var_dump($response->getHeaders());
```

Comment:
I think it can be confusing that we show two types of headers in this example (request and response).
```suggestion
     * var_dump($response);
```